src/axom/inlet/LuaReader.cpp

```diff
@@ -64,114 +54,107 @@ bool LuaReader::parseString(const std::string& luaString)
-  {
-    SLIC_WARNING(fmt::format("Variable cannot end with scope delimiter: {0}",
-                             id));
-    return false;
-  }
 
-  bool atGlobalScope = true;
-  std::vector<std::string> tokens;
-  axom::utilities::string::split(tokens, temp_id, SCOPE_DELIMITER);
+bool LuaReader::getDouble(const std::string& id, double& value)
+{
+  return getValue(id, value);
+}
 
-  // Clear the lua stack because we always call with fully qualified names
-  lua_settop(m_luaState, 0);
-  for (std::string token : tokens)
-  {
-    if(atGlobalScope)
-    {
-      lua_getglobal(m_luaState, token.c_str());
-    }
-    else
-    {
-      lua_getfield(m_luaState, -1, token.c_str());
-    }
-    if(lua_isnil(m_luaState, -1))
-    {
-      // variable not found
-      return false;
-    }
-    atGlobalScope = false;
-  }
 
-  return true;
+bool LuaReader::getInt(const std::string& id, int& value)
+{
+  return getValue(id, value);
 }
 
 
-bool LuaReader::getBool(const std::string& id, bool& value)
+bool LuaReader::getString(const std::string& id, std::string& value)
 {
-  if (!findVariable(id))
-  {
-    return false;
-  }
-  value = (bool)lua_toboolean(m_luaState, -1);
-  return true;
+  return getValue(id, value);
 }
 
+bool LuaReader::getIntMap(const std::string& id, std::unordered_map<int, int>& values) {
+  return getMap(id, values, sol::type::number);
+}
 
-bool LuaReader::getDouble(const std::string& id, double& value)
-{
-  if (!findVariable(id))
-  {
-    return false;
-  }
-  value = (double)lua_tonumber(m_luaState, -1);
-  return true;
+bool LuaReader::getDoubleMap(const std::string& id, std::unordered_map<int, double>& values) {
+  return getMap(id, values, sol::type::number);
 }
 
+bool LuaReader::getBoolMap(const std::string& id, std::unordered_map<int, bool>& values) {
+  return getMap(id, values, sol::type::boolean);
+}
+ 
+bool LuaReader::getStringMap(const std::string& id, std::unordered_map<int, std::string>& values) {
+  return getMap(id, values, sol::type::string);
+}
 
-bool LuaReader::getInt(const std::string& id, int& value)
-{
-  if (!findVariable(id))
-  {
+template <typename T>
+bool LuaReader::getValue(const std::string& id, T& value) {
+  std::vector<std::string> tokens;
+  axom::utilities::string::split(tokens, id, SCOPE_DELIMITER);
+  if (tokens.size() == 1 && m_lua[id].valid()) {
+    value = m_lua[id];
+    return true;
+  }
+  
+  if (!m_lua[tokens[0]].valid()) {
     return false;
   }
-  value = (int)lua_tonumber(m_luaState, -1);
-  return true;
+  sol::table t = m_lua[tokens[0]];
+  for (size_t i = 1; i < tokens.size()-1; i++) {
+    if (t[tokens[i]].valid()) {
+      t = t[tokens[i]];
+    } else {
+      return false;
+    }
+  }
+  if (t[tokens.back()].valid()) {
+    value = t[tokens.back()];
+    return true;
+  }
+  
+  return false;
 }
 
+template <typename T>
+bool LuaReader::getMap(const std::string& id, std::unordered_map<int, T>& values, sol::type type) {
```

Comment:
Do we always want to extract the data to an explicit map?
Are there potential issues when `T` is not a native type (i.e. int, string, double,bool) ?

Reply:
As of now, we don't provide an interface for extracting non-native types.